### DISCUSSION TITLE
How would you solve ambiguity with signature and hover eldoc fns?

### DISCUSSION THREAD

**Discussion by zikajk** [joaotavora/eglot#1240] [Q&A]
Hi, when i move cursor over Clojure fn I get on-hover and signature documentation in eldoc buffer. Usually I can live with only one of them as they can be the same.
It seems to be specific for clojure-lsp, i dont have this problem with other languages i have tried.
What are the possible solutions to this "problem"? 
I was thinking about enabling on-hover for mouse only with clojure-mode but i don't know if it is possible...
![image](https://github.com/joaotavora/eglot/assets/40757859/882f89ab-c0a6-4b84-b67d-a623d166872f)

**Comment by joaotavora** [comment]
> I was thinking about enabling on-hover for mouse only with clojure-mode but i don't know if it is possible...

I think I understand what you mean, but I don't know if it's possible.  What exactly do you mean "mouse only"? You mean hovering with the mouse on a specific symbol code without ever pressing `mouse-1`?

**Reply by zikajk** [reply]
Yes, cider is doing something similar with pos-tip. It shows fn docstring when you point over it. 
But i would love to have some more elegant solution like - if signature and on-hover are not different (or smth like that, in fact the first line of on-hover and signature are different), show only the prefered one - but i don't know how to achieve that.

**Reply by joaotavora** [reply]
OK, but you didn't quite answer my question:

> You mean hovering with the mouse on a specific symbol [in] code without ever pressing mouse-1?

Do you understand the question?  If so please answer it, if not I can clarify.  The feasibility of this enhancement depends on your reply.

**Reply by zikajk** [reply]
I said yes and that Cider was doing something similar if you wanted a demonstration.
If that is not enough then i don't get the question :-).

**Reply by joaotavora** [reply]
OK, then I would say that is hard to do, and probably undesirable, it would mean putting `help-echo` properties in basically all the places in the buffer after asking the LSP server where the document's symbols are. It's similar to a jit-lock-function, and those aren't trivial to get working well.

I'd say Eldoc is not the tool for that, as Eldoc is designed as an **at-point** documentation system, and simply hovering doesn't count as "at point".

On the other hand (but according to your reply, it's not what you want), it is possible and relatively easy to do different behaviour by checking if command was a `mouse-set-point`, which is triggered with `mouse-1`.

**Reply by zikajk** [reply]
Okay, let's see if someone hasn't sorted this out. 
Unfortunately, most people who write Clojure use LSP-Mode.
